### PR TITLE
mount-setup: minor tweaks

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/shared/mount-setup.c [hunk 1]
```diff
@@ -211,19 +211,16 @@ static int mount_one(const MountPoint *p, bool relabel) {
...
-                int j;
-
-                j = mount_one(mount_table + i, loaded_policy);
-                if (j != 0 && r >= 0)
-                        r = j;
+        for (size_t i = 0; i < n; i ++) {
+                r = mount_one(mount_table + i, loaded_policy);
+                if (r != 0 && ret >= 0)
+                        ret = r;
         }
```
Comment (YHNdnzj) [inline src/shared/mount-setup.c]: You can use `FOREACH_ARRAY` here

**Review by YHNdnzj — APPROVED** [review]

File: src/shared/mount-setup.c (hunk 1 above)
Comment (poettering) [reply src/shared/mount-setup.c]: indeed. fixed.

File: src/shared/mount-setup.c [hunk 2]
```diff
@@ -205,27 +205,26 @@ static int mount_one(const MountPoint *p, bool relabel) {
...
+                r = mount_one(mp, loaded_policy);
+                if (r != 0 && ret >= 0)
+                        ret = r;
         }
 
-        return r;
+        return ret;
 }
 
 int mount_setup_early(void) {
```
Comment (YHNdnzj) [inline src/shared/mount-setup.c:230]: The `assert_cc` below is redundant now?

Comment (poettering) [reply src/shared/mount-setup.c:230]: hmm? what do you mean? the new assert() in mount_points_setup() just checks if the specific invocation makes sense, while the new assert_cc() the patch adds just checks that the definition of N_EARLY_MOUNTS roughly makes sense. That's all.

I mean, all asserts are by definition redundant, they are just safeguards and encode assumptions. While the assert() and the assert_cc() here might check similar things, there's still quite a major semantical difference between "check a definition for validity" and "check a function invocation for validity".

Comment (YHNdnzj) [reply src/shared/mount-setup.c:230]: I mean that you can remove https://github.com/systemd/systemd/pull/27647/files#diff-167b86cba7662f37f02f27530b191527b457bb8755c65163bd167fded58ffc49R231 :)

It's the same as the newly-added `assert_cc`

**Review by yuwata — APPROVED** [review]

File: src/shared/mount-setup.c (hunk 2 above)
Comment (poettering) [reply src/shared/mount-setup.c:230]: indeed!

**Pull request merged**